docs/book/v1/migrate-implements-factory-interface-to-use-psr-container.md

```diff
@@ -0,0 +1,95 @@
+# Migrate Class Implements FactoryInterface to use Psr Container
+
+To migrate existing factory classes that implements `Laminas\ServiceManager\Factory\FactoryInterface`, there is a Rector rule that can be used to migrate for that. For example, we have the following factory class:
+
+```php
+use My\Service;
+use Interop\Container\ContainerInterface;
+use Laminas\ServiceManager\Factory\FactoryInterface;
+
+class ImplementsRootAbstractFactoryInterface implements FactoryInterface
+{
+    public function __invoke(ContainerInterface $container, $requestedName, array $options = null)
+    {
+        return new Service();
+    }
+}
+```
+
+We need to migrate to :
+
+```php
+use My\Service;
+use Interop\Container\ContainerInterface;
+use Laminas\ServiceManager\Factory\FactoryInterface;
+
+class ImplementsRootAbstractFactoryInterface
+{
+    public function __invoke(\Psr\Container\ContainerInterface $container)
+    {
+        return new Service();
+    }
+}
+```
```

Comment:
This is not the real goal, only an intermediate step. 

The goal should be:

```php
use My\Service;
use Laminas\ServiceManager\Factory\FactoryInterface;
use Psr\Container\ContainerInterface;

class ServiceFactory
{
    public function __invoke(ContainerInterface $container)
    {
        return new Service();
    }
}
```

Better:

```php
use Laminas\ServiceManager\Factory\FactoryInterface;
use My\Service;
use Psr\Container\ContainerInterface;

final class ServiceFactory
{
    public function __invoke(ContainerInterface $container): Service
    {
        return new Service();
    }
}
```

We should describe all the steps to achieve this goal in order to show the whole workflow. (All used standard Rector rules should be linked with the original documentation.)

Suggestion for the steps:

1. Remove the Interface from the Factory
1. Additional Adjustments
   1. Auto Import Names
   2. Finalize Factory Class
   3. Add Return Type

And add a note on how to sort the use statements via CS tools.

Reply:
@froschdesign add return type hint is not the original issue you wrote at https://github.com/laminas/laminas-servicemanager-migration/issues/34

There is TypeDeclaration rector rule for that, but out of this documentation scope, possibly can be noted to link, but not mandatory.

https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#typedeclaration

Reply:
> There is TypeDeclaration rector rule for that, but out of this documentation scope, possibly can be noted to link, but not mandatory.

Right, that's a standard rule of Rector, but that doesn't mean we have to show only half an example in documentation. Therefore I suggested the headline "Additional Adjustments".
And if a standard rule is used then also a link must be added.

Reply:
Ok, updated with add step + additional adjustment https://github.com/laminas/laminas-servicemanager-migration/pull/35/commits/544547babdc7787d9cc1333a0f3cddbe87885089